ui/pages/confirmations/confirmation/alerts/setAlerts.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+import {
+  Alert,
+  clearAlerts,
+  updateAlerts,
+} from '../../../../ducks/confirm-alerts/confirm-alerts';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+
+const setAlerts = (alertOwnerId: string) => {
+  const dispatch = useDispatch();
+  const t = useI18nContext();
+  const alerts: Alert[] = useMemo(() => [], [t]);
```

Comment:
Is this the injection point where the template confirmations would be generated?

For the existing inline alerts, they use an internal `alertStateReducer` and `useAlertState` hook which just extracts the alerts from the current template, and uses the reducer to record dismissal.

Is the plan in a future PR to migrate those here, so the inline alerts also use the new context and hook, or are we keeping them as two distinct systems for now for the sake of time? One for inline alerts, and one for modal alerts?

They'd don't strictly need merging now, but perhaps worth doing in future.

Reply:
I agree to different system of alerts is not great, I think migrating all old alerts will be out of scope for this PR. I will create separate task to migrate old alerts, we can prioritise it during planning.